File: library_management/library_management/report/school_result_color_report/school_result_color_report.py
# Copyright (c) 2025, shridevi and contributors
# For license information, please see license.txt

# import frappe

import frappe
def execute(filters=None):
    columns = [
        {"label": "Student", "fieldname": "student", "fieldtype": "Data", "width": 200},
        {"label": "Positive Percentage", "fieldname": "positive_percentage", "fieldtype": "Float", "width": 150},
        {"label": "Negative Percentage", "fieldname": "negative_percentage", "fieldtype": "Float", "width": 150},
    ]
    data = []
    results = frappe.get_all("School result", fields=["student", "percentage"])
    for r in results:
        row = {
            "student": r.student,
            "positive_percentage": r.percentage if r.percentage >= 0 else None,
            "negative_percentage": r.percentage if r.percentage < 0 else None,
        }
        data.append(row)
    return columns, data

File: library_management/library_management/report/school_result_color_report/school_result_color_report.js
// Copyright (c) 2025, shridevi and contributors
// For license information, please see license.txt
frappe.query_reports["School Result Color Report"] = {
    onload: function(report) {
        report.page.set_title("School Result Color Report");
    },
    formatter: function(value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);
        if (column.fieldname === "positive_percentage" && data.positive_percentage !== undefined && data.positive_percentage !== null) {
            value = `<span style="color:green; font-weight:bold;">${data.positive_percentage}</span>`;
        }

        if (column.fieldname === "negative_percentage" && data.negative_percentage !== undefined && data.negative_percentage !== null) {
            value = `<span style="color:red; font-weight:bold;">${data.negative_percentage}</span>`;
        }
        return value;
    }
};


